File: leetcode-14/python/leetcode-14.py
class Solution:
    def longestCommonPrefix(self, strs):
        """

        :param strs:
        :return: str
        """
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]

        temp = strs[0]
        temp_strs = strs[1:]
        ret = ""
        for i in range(len(temp)):
            print(i)
            for j in temp_strs:
                if i > len(j)-1 or temp[i] != j[i]:
                    return ret
            ret += temp[i]
        return ret
    def longestCommonPrefix2(self, strs):
        if len(strs) == 0:
            return ""
        res = ""
        for each in zip(*strs):
            if len(set(each)) == 1:
                res += each[0]
            else:
                return res
        return res

if __name__ == '__main__':
    s = Solution()

    result = s.longestCommonPrefix2(["flower", "flow"])

    result = s.longestCommonPrefix2(["a", "aa", "aaa"])
    print(result)

File: leetcode-14/javascript/leetcode-14.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  if (strs.length == 0) {
    return '';
  }
  if (strs.length == 1) {
    return strs[0];
  }
  let temp = strs[0];
  let result = '';
  for (let i = 0; i < temp.length; ++i) {
    for (let j = 1; j < strs.length; ++j) {
      if (i >= strs[j].length || temp[i] !== strs[j][i]) {
        return result;
      }
    }
    result += temp[i];
  }
  return result;
};

let ret = longestCommonPrefix(['flower', 'fl']);
console.log(ret);
ret = longestCommonPrefix(['aaa', 'aa', 'aaa']);
console.log(ret);

ret = longestCommonPrefix(["aca", "cba"]);
console.log(ret);